firebase-firestore/src/main/java/com/google/firebase/firestore/local/IndexBackfiller.java

```diff
@@ -0,0 +1,110 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+import androidx.annotation.Nullable;
+import com.google.firebase.firestore.util.AsyncQueue;
+import java.util.concurrent.TimeUnit;
+
+public class IndexBackfiller {
+  /** How long we wait to try running LRU GC after SDK initialization. */
+  private static final long INITIAL_GC_DELAY_MS = TimeUnit.MINUTES.toMillis(1);
+  /** Minimum amount of time between GC checks, after the first one. */
+  private static final long REGULAR_GC_DELAY_MS = TimeUnit.MINUTES.toMillis(5);
+
+  public static class Results {
+    private final boolean hasRun;
+
+    private final int entriesAdded;
+    private final int entriesRemoved;
+
+    static IndexBackfiller.Results DidNotRun() {
+      return new IndexBackfiller.Results(/* hasRun= */ false, 0, 0);
+    }
+
+    Results(boolean hasRun, int entriesAdded, int entriesRemoved) {
+      this.hasRun = hasRun;
+      this.entriesAdded = entriesAdded;
+      this.entriesRemoved = entriesRemoved;
+    }
+
+    public boolean hasRun() {
+      return hasRun;
+    }
+
+    public int getEntriesAdded() {
+      return entriesAdded;
+    }
+
+    public int getEntriesRemoved() {
+      return entriesRemoved;
+    }
+  }
+
+  public class Scheduler implements StartStopScheduler {
```

Comment:
I am trying to figure out what functionality a StartStopSchedule offers over a Scheduler, but I am not convinced I will ever find out :)

Reply:
Renamed the base class to Scheduler. To be honest, I'm not sure why the LruGarbageCollector's Scheduler used the abstract base class to begin with since it's not shared with any other class.